static/copo/js/generic_handlers.js

```diff
@@ -4031,24 +4031,32 @@ function initialiseComponentDropdownMenu() {
     placeholder: 'Choose an option',
     allowClear: false,
     templateSelection: function (data) {
-      if (!data || !data.id || !data.text) return data.text || '';
+      if (!data || !data.id || !data.text) return data.text || ''; // Placeholder or empty
+
       // Check if the selected option begins with an asterisk (*)
       // If yes, render a red asterisk and display the rest of the text
-      if (/^\*\s*/.test(data.text)) {
-        const text = data.text.replace(/^\*\s*/, '');
-        return `<span class="asterisk" aria-hidden="true">*</span>${text}`;
+      const $el = $(data.element);
+      const $selection = $('<span>').text(data.text);
+
+      if ($el.data('asterisk')) {
+        $selection.attr('data-asterisk', 'true');
```

Comment:
You have already added "data-asterisk" as an attribute in copo_single_cell.js, why you handle it here again?